From 9ebddac7ea2a1f4b4ce3335a78312a58dfaadb4d Mon Sep 17 00:00:00 2001
From: "Luck, Tony" <tony.luck@intel.com>
Date: Fri, 8 Nov 2013 14:03:33 -0800
Subject: ACPI, x86: Fix extended error log driver to depend on
 CONFIG_X86_LOCAL_APIC
Git-commit: 9ebddac7ea2a1f4b4ce3335a78312a58dfaadb4d
References: fate#315508
Patch-mainline: 3.13-rc1

Randconfig build by Fengguang's robot army reported:

   drivers/built-in.o: In function `extlog_print':
   >> acpi_extlog.c:(.text+0xcc719): undefined reference to `boot_cpu_physical_apicid'

The config had CONFIG_SMP=n so we picked up this definition from:

  <asm/cpu.h>: #define cpu_physical_id(cpu) boot_cpu_physical_apicid

But boot_cpu_physical_apicid is defined in arch/x86/kernel/apic/apic.c
which is only built if CONFIG_X86_LOCAL_APIC=y.

Reported-by: Fengguang Wu <fengguang.wu@intel.com>
Signed-off-by: Tony Luck <tony.luck@intel.com>
Cc: Chen Gong <gong.chen@linux.intel.com>
Cc: Rafael J. Wysocki <rjw@rjwysocki.net>
Link: http://lkml.kernel.org/r/6be3afdcad7968f7fb7c0b681e547b3e872e44dd.1383947368.git.tony.luck@intel.com
Signed-off-by: Ingo Molnar <mingo@kernel.org>
Acked-by: Borislav Petkov <bp@suse.de>
---
 drivers/acpi/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/acpi/Kconfig b/drivers/acpi/Kconfig
index 08eadb4a57cb..e11faae81ed9 100644
--- a/drivers/acpi/Kconfig
+++ b/drivers/acpi/Kconfig
@@ -374,7 +374,7 @@ source "drivers/acpi/apei/Kconfig"
 
 config ACPI_EXTLOG
 	tristate "Extended Error Log support"
-	depends on X86_MCE
+	depends on X86_MCE && X86_LOCAL_APIC
 	select EFI
 	select UEFI_CPER
 	default n

